runtime/binding-ws/src/main/java/io/aklivity/zilla/runtime/binding/ws/internal/stream/WsServerFactory.java

```diff
@@ -215,25 +209,58 @@ else if (key != null &&
                     if (newRoute != null && (route == null || newRoute.order < route.order))
                     {
                         route = newRoute;
-                        protocol = newProtocol;
+                        websocketProtocol = newProtocol;
                     }
                 }
 
-                if (route != null)
+                if (route != null && WEBSOCKET_VERSION_13.equals(version))
                 {
-                    newStream = new WsServer(
-                        sender,
-                        originId,
-                        routedId,
-                        initialId,
-                        route.id,
-                        key,
-                        protocol,
-                        scheme,
-                        authority,
-                        path)::onNetMessage;
+                    final String key = headers.get("sec-websocket-key");
+                    if (key != null &&
+                        WEBSOCKET_UPGRADE.equalsIgnoreCase(upgrade))
+                    {
+                        // HTTP1.1 handshake
+                        newStream = new WsServer(
+                            sender,
+                            originId,
+                            routedId,
+                            initialId,
+                            route.id,
+                            HttpVersion.HTTP_1_1,
+                            key,
+                            websocketProtocol,
+                            scheme,
+                            authority,
+                            path)::onNetMessage;
+                    }
+                    else if (WEBSOCKET_PROTOCOL.equalsIgnoreCase(protocol))
+                    {
+                        // HTTP2 handshake
+                        newStream = new WsServer(
+                            sender,
+                            originId,
+                            routedId,
+                            initialId,
+                            route.id,
+                            HttpVersion.HTTP_2,
+                            null,
+                            websocketProtocol,
+                            scheme,
+                            authority,
+                            path)::onNetMessage;
+                    }
```

Comment:
Given the vastly different approaches to the handshake, suggest you avoid all the `if/else` logic and instead have separate `Ws11Server` and `Ws2Server`, both extending generic abstract `WsServer` if there is enough overlap in implementation beyond the handshake.